presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractRowBlock.java

```diff
@@ -107,6 +108,27 @@ public long getRegionSizeInBytes(int position, int length)
         return regionSizeInBytes;
     }
 
+    @Override
+    public long getPositionsSizeInBytes(boolean[] positions)
+    {
+        checkValidPositions(positions, getPositionCount());
+
+        int usedPositionCount = 0;
+        long sizeInBytes = 0;
+        for (int i = 0; i < positions.length; i++) {
+            if (positions[i]) {
+                usedPositionCount++;
+                int startFieldBlockOffset = getFieldBlockOffset(i);
+                int endFieldBlockOffset = getFieldBlockOffset(i + 1);
+                int fieldBlockLength = endFieldBlockOffset - startFieldBlockOffset;
+                for (int j = 0; j < numFields; j++) {
+                    sizeInBytes += getRawFieldBlocks()[j].getRegionSizeInBytes(startFieldBlockOffset, fieldBlockLength);
```

Comment:
Don't do this. This is the same `O(n^2)` bug you are trying to fix here, just in a different kind of `Block`. Ditto for `getRegionSizeInBytes` implementation of this Block.